### ISSUE TITLE
Support for AWS::LanguageExtensions

### ISSUE THREAD

**Issue by shaunster666** [GreenfieldTech/cloudformation-tool#4]
Hi,

Does the current version support AWS::LanguageExtensions?

If not, could this be made available at some point?

Many  thanks for making cftool public, much appreciated

**Comment by shaunster666** [comment]
To facilitate this all cftool has to do is ignore this instruction but include it in the compiled stack:

Transform: AWS::LanguageExtensions

**Comment by guss77** [comment]
As far as I've tested - cloudformation-tool does not interfere with using `AWS::LanguageExtensions` transform.

I've just checked with the with [the sample from the guide](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/transform-aws-languageextensions.html) and it deploys correctly with `cftool` and shows the demo function working. For example, using this command:

```
cftool --profile myprofile create \
    -p QueueList=a,b,c -p QueueNameParam=test-queue \
    ./samples/tests/aws-language-extensions-demo.yaml test-language-extensions
```

Generally, this gem does not change things it doesn't understand - except to transform them through Ruby's Psyche library (which we use the read the YAML and then to right it back out).

 If you have an issue where `cftool` messes up usage of some feature - I'd appreciate it if you can show a (hopefully minimal) test case so I can try it out and fix the problem.

**Comment by shaunster666** [comment]
Many thanks for your advice.

If I move AWS::LanguageExtensions transform from an include file to the parent .cf file like so it works fine:


---
Description: test
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::LanguageExtensions'


Parameters:

  Env:
    Type: String
    Description: Define environment


Include:

  - shared/includes/conditions.yaml
  - shared/includes/waf.yaml

**Comment by guss77** [comment]
I see - I didn't test that. When putting `Transform` into an included file, I can see that `cftool` throws an exception: `cloud_formation.rb:134:in 'merge!': no implicit conversion of String into Hash (TypeError)`.

@shaunster666 If you would have mentioned that, it would have made it much clearer that there's an issue here.

**Comment by guss77** [comment]
This issue is fixed in release 1.5.14.

I've started adding sample stacks to record issues as part of the future test suite. You can see this issue is fixed by running `cftool compile samples/tests/aws-language-extensions-stack` in the repository.
